### PR TITLE
Automate RSIC-V compliance check with github actions

### PR DESCRIPTION
**Pull request by ekiwi**
Looks like this: https://github.com/ekiwi/serv/actions/runs/164021731

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  compliance:
+    name: RISC-V Compliance Test
+    runs-on: ubuntu-20.04
+    env:
+      SERVANT_VERSION: 1.0.2
```
Comment (ekiwi) [inline .github/workflows/ci.yml:14]: @olofk is there any way to run a command to get the latest version of sevant? Maybe through fusesoc?

**Comment by olofk** [comment]
Ok, this is absolutely fantastic and I have no idea how I missed this. As it has been a while since this PR was submitted I just wanted to check if it's still valid or if something needs to be changed.

I see the problem with SERVANT_VERSION and I think it can be solved in another way by adding `--build-root=some/known/directory` to have this part determinstic

**Comment by ekiwi** [comment]
> As it has been a while since this PR was submitted I just wanted to check if it's still valid or if something needs to be changed.

I can try to find some time this week or next week to rebase on the latest master branch and try to see if anything changed.

**Comment by ekiwi** [comment]
> I see the problem with SERVANT_VERSION and I think it can be solved in another way by adding `--build-root=some/known/directory` to have this part determinstic

What would I pass `--build-root` to? I get the following error if I just try to pass it to `fusesoc run`.

```
 fusesoc run servant_1.0.2: error: unrecognized arguments: --build-root=servant_x
```

**Comment by olofk** [comment]
Sorry. `--build-root` should go between the `run` and `servant` parameters because it's a command parameter rather than a target parameter.

I took the liberty to change this myself in your commit, as well as fixing up a few other things that had been deprecated since you originally wrote this. The end result is that it now runs fine! Many thanks for your contribution

**Pull request closed without merging**